Added: Antiderivative (curve, surface & volume)

VikingScientist opened the pull request:
Fixes #187.

For review it might be interesting to go through a20feb1024a88ae7bbcc2d334d500e1a506ce1c3 which addeded `agents.md` file

VikingScientist committed:
Added: curve.antiderivative

VikingScientist committed:
AI cleanup

VikingScientist committed:
Added:_surface.antiderivative

VikingScientist committed:
Added: volume.antiderivative

VikingScientist committed:
Refactor: moved get_antiderivative_spline to splineobject.py

VikingScientist committed:
Cosmetics: whitespace fixes

TheBB commented:
Why are the two first commits here (uv link and agents.md)?

Agents.md was in your other PR. The UV link fix is already on master.

TheBB commented:
I wonder if we need the full docstring in the functions `get_antiderivative_{curve|surface|volume}`. They are really just type-modified wrappers of `get_antiderivative_spline` and they work exactly the same way. I think the docstring should reflect that.

VikingScientist commented:
> Why are the two first commits here (uv link and agents.md)?
> 
> Agents.md was in your other PR. The UV link fix is already on master.

Rebased on master. Fixed now


> I wonder if we need the full docstring in the functions get_antiderivative_{curve|surface|volume}. They are really just type-modified wrappers of get_antiderivative_spline and they work exactly the same way. I think the docstring should reflect that.

While I agree that it is a lot of copy-paste, it does however not appear in my text editor (vscode) as a helpful text if it is not included in the subclasses. See attached screenshot:


<img width="1034" height="411" alt="Screenshot from 2026-01-02 14-18-12" src="https://github.com/user-attachments/assets/c196baaf-e32b-4b3a-8da6-0ce2787d30b0" />

vs if I do include it as documentation in the submethods:


<img width="1031" height="442" alt="Screenshot from 2026-01-02 14-24-44" src="https://github.com/user-attachments/assets/887a1041-af66-4008-8cfd-2cc26bf5107d" />

It also has the added benefit of including dimension-specific code examples in the documentation. Minor stuff such as no need for direction parameter for a curve call.

VikingScientist committed:
Documentation fixes for get_antiderivative_spline
